src/components/common/SettingsPage.tsx

```diff
@@ -0,0 +1,74 @@
+import {
+  Button,
+  Container,
+  Group,
+  Stack,
+  Text,
+  TextInput,
+  Title,
+} from "@mantine/core";
+import { notifications } from "@mantine/notifications";
+import { useState } from "react";
+import { useCurrentWorkspace } from "@/hooks/workspaces/useCurrentWorkspace";
+import { WorkspaceClient } from "@/models/Workspace/WorkspaceClient";
+
+export function SettingsPage(): JSX.Element {
+  const workspace = useCurrentWorkspace();
+  const [workspaceName, setWorkspaceName] = useState(workspace.name);
+
+  const [saveWorkspace, isWorkspaceSaving] = WorkspaceClient.useUpdate({
+    onSuccess: () => {
+      notifications.show({
+        title: "Workspace name updated",
+        message: "The workspace name was saved successfully.",
+        color: "green",
+      });
+    },
+    onError: (error) => {
+      notifications.show({
+        title: "Failed to update workspace name",
+        message: error.message,
+        color: "red",
+      });
+    },
+  });
+
+  const handleSave = () => {
```

Comment:
I haven't written a style guide doc yet but one thing I like remaining consistent on is naming all event handlers with the `on` prefix, and typically naming them after the user action that triggers it, rather than the business logic it performs.

So in this case, `onSubmit` or `onSubmitClick` would be preferable names.